src/viam/logging.py

```diff
@@ -53,13 +85,34 @@ def addHandlers(logger: logging.Logger):
     # filter out logs at error level or above
     handler.setLevel(LOG_LEVEL)
     handler.addFilter(filter=lambda record: (record.levelno < ERROR))
-    logger.addHandler(handler)
 
     err_handler = logging.StreamHandler(stream=sys.stderr)
     err_handler.setFormatter(format)
     # filter out logs below error level
     err_handler.setLevel(max(ERROR, LOG_LEVEL))
-    logger.addHandler(err_handler)
+
+    if _MODULE_PARENT is not None and not default:
+        _startModuleLogging(logger)
+    else:
+        logger.addHandler(handler)
+        logger.addHandler(err_handler)
+
+
+def _startModuleLogging(logger: logging.Logger):
+    handler = ModuleHandler(logger)
+
+    format = ColorFormatter("%(asctime)s\t\t" + "%(levelname)s\t" + "%(name)s (%(filename)s:%(lineno)d)\t" + "%(message)s\t")
+    handler.setFormatter(format)
+    handler.setLevel(LOG_LEVEL)
+
+    logger.addHandler(handler)
+
+
+def setParent(parent):
+    global _MODULE_PARENT
+    _MODULE_PARENT = parent
```

Comment:
Nice! I like the use of the global and just applying it to all existing loggers. What type are `parent` and `_MODULE_PARENT`/should we add type annotations for them?

Reply:
Yeah, I was struggling with this too. The type here is a `RobotClient`. But I can't import it because the class also imports `logging`. Thanks for reminding me to flag this for further discussion!